File: Backjoon/Python/13417_카드 문자열.py
"""
 카드 문자열
 https://www.acmicpc.net/problem/13417
"""

for _ in range(int(input())):
    n = int(input())
    cards = input().split()
    result = cards[0]
    for i in range(1,n):
        result = cards[i] + result if cards[i] <= result[0] else result + cards[i]
    print(result)

File: Backjoon/Sliver/13417_카드 문자열.swift
/*
 카드 문자열
 https://www.acmicpc.net/problem/13417
*/

import Foundation

(0..<Int(readLine()!)!).forEach { _ in
    let n = Int(readLine()!)!
    var input = readLine()!.components(separatedBy: " ")
    var result = input.first!
    for i in 1..<n {
        result = input[i] <= "\(result.first!)" ? input[i] + result : result + input[i]
    }
    print(result)
}

